dask_jobqueue/tests/test_flux.py

```diff
@@ -0,0 +1,242 @@
+            w = workers[0]
+            assert w["memory_limit"] == 2 * 1024**3
+            assert w["nthreads"] == 2
+
+            cluster.scale(0)
+
+            start = time()
+            while client.scheduler_info()["workers"]:
+                sleep(0.100)
+                assert time() < start + QUEUE_WAIT
+
+
+@pytest.mark.env("flux")
+def test_adaptive(loop):
+    with FluxCluster(
+        walltime="00:02:00",
+        cores=2,
+        processes=1,
+        memory="2GB",
+        loop=loop,
+    ) as cluster:
+        cluster.adapt()
+        with Client(cluster) as client:
+            future = client.submit(lambda x: x + 1, 10)
+
+            start = time()
+            client.wait_for_workers(1, timeout=QUEUE_WAIT)
+
+            assert future.result(QUEUE_WAIT) == 11
+
+            del future
+
+            start = time()
+            while client.scheduler_info()["workers"]:
+                sleep(0.100)
+                assert time() < start + QUEUE_WAIT
+
+
+@pytest.mark.env("flux")
+def test_config_name_flux_takes_custom_config():
+    conf = {
+        "queue": "myqueue",
+        "account": "myaccount",
+        "ncpus": 1,
+        "cores": 1,
+        "memory": "2 GB",
+        "walltime": "00:02",
+        "job-extra": None,
+        "job-extra-directives": [],
+        "name": "myname",
+        "processes": 1,
+        "interface": None,
+        "death-timeout": None,
+        "local-directory": "/foo",
+        "shared-temp-directory": None,
+        "extra": None,
+        "worker-extra-args": [],
+        "env-extra": None,
+        "job-script-prologue": [],
+        "log-directory": None,
+        "shebang": "#!/usr/bin/env bash",
+        "job-cpu": None,
+        "job-mem": None,
+    }
+
+    with dask.config.set({"jobqueue.flux.config-name": conf}):
+        with FluxCluster(config_name="flux.config-name") as cluster:
+            assert cluster.job_name == "myname"
+
+
+@pytest.mark.env("flux")
+def test_worker_name_uses_cluster_name(loop):
+    # The environment variable setup below is similar to a job array setup
+    # where you would use SLURM_ARRAY_JOB_ID to make sure that Dask workers
+    # belonging to the same job array have different worker names
+    with FluxCluster(
+        cores=1,
+        memory="2GB",
+        name="test-$MY_ENV_VARIABLE",
+        job_script_prologue=["MY_ENV_VARIABLE=my-env-variable-value"],
+        loop=loop,
+    ) as cluster:
+        with Client(cluster) as client:
+            cluster.scale(jobs=2)
+            print(cluster.job_script())
+            client.wait_for_workers(2, timeout=QUEUE_WAIT)
+            worker_names = [
+                w["id"] for w in client.scheduler_info()["workers"].values()
+            ]
+            assert sorted(worker_names) == [
+                "test-my-env-variable-value-0",
+                "test-my-env-variable-value-1",
+            ]
+
+
+@pytest.mark.env("flux")
```

Comment:
Are there some tests which can be run without spawning a complete Flux cluster? If so, they should be without this fixture I think?

Reply:
Maybe if we mocked it? But it seems easier to just spin up a container base with flux. Let me know what you have in mind specifically though!